src/store/juju/selectors.ts

```diff
@@ -500,32 +501,36 @@ export const hasModels = createSelector(
   (modelList) => Object.keys(modelList).length > 0,
 );
 
-export function getModelWatcherDataByUUID(modelUUID: string) {
-  return createSelector(getModelWatcherData, (modelWatcherData) => {
+export const getModelWatcherDataByUUID = createSelector(
+  [getModelWatcherData, (_state: RootState, modelUUID: string) => modelUUID],
+  (modelWatcherData, modelUUID) => {
     if (modelWatcherData?.[modelUUID]) {
       return modelWatcherData[modelUUID];
     }
+
     return null;
-  });
-}
+  },
+);
 
-export function getModelInfo(modelUUID: string) {
-  return createSelector(
-    getModelWatcherDataByUUID(modelUUID),
-    (modelWatcherData): WatcherModelInfo | null => {
-      if (modelWatcherData) {
-        return modelWatcherData.model;
-      }
-      return null;
-    },
-  );
-}
+export const getModelInfo = createSelector(
+  getModelWatcherDataByUUID,
+  (modelWatcherData): WatcherModelInfo | null => {
+    if (modelWatcherData) {
+      return modelWatcherData.model;
+    }
 
-export function getModelUUIDFromList(
-  modelName?: string | null,
-  ownerName?: string | null,
-) {
-  return createSelector(getModelList, (modelList: ModelsList) => {
+    return null;
+  },
+);
+
+export const getModelUUIDFromList = createSelector(
+  [
+    getModelList,
+    (_state: RootState, modelName?: string | null) => modelName || null,
+    (_state: RootState, _modelName, ownerName?: string | null) =>
+      ownerName || null,
+  ],
+  (modelList: ModelsList, modelName, ownerName) => {
```

Comment:
If you wanted you could pass both modelName and ownerName in an object:

```suggestion
    (_state: RootState, modelName, ownerName?: string | null) =>
      { modelName, ownerName },
  ],
  (modelList: ModelsList, { modelName, ownerName }) => {
```

Reply:
I'm not sure if this is actually what we want. Reselect appears to use references returned from the selectors to manage it's caching. Returning an object from the selector would violate this.

Reply:
Ah yes that's a very good point. Nevermind then.